### PR TITLE
feat(epic-b): filter non-diff file comments to reduce noise

### PR DESCRIPTION
**Pull request by RC918**
## Summary

Implements EPIC B optimization to filter out review comments for files NOT in the PR diff. This reduces noise from pre-existing issues in unchanged files, allowing reviewers to focus on the actual changes.

**Changes:**
- Add `REVIEWER_FILTER_NON_DIFF_FILES` feature flag (default: `true`) in settings.py
- Implement `filter_non_diff_file_comments()` function in review_comment_schema.py
- Integrate filter into `publisher_node` with telemetry tracking
- Add transparency note in review body when comments are filtered (e.g., "*Note: 5 comments for 2 files not in this PR diff were filtered...*")

## Review & Testing Checklist for Human

- [ ] **Verify default=True is intentional** - The feature is ON by default, meaning it will immediately start filtering comments. If this is too aggressive, consider setting default to `false` for gradual rollout.
- [ ] **Test with a real PR** - Create/find a PR where the AI reviewer generates comments for files not in the diff, and verify:
  - Comments for non-diff files are filtered out
  - Transparency note appears in the review body
  - Telemetry logs show `non_diff_filtered_count` and `filtered_files`
- [ ] **Test edge cases** - Verify behavior when:
  - All comments are for non-diff files (should show only the note)
  - No comments are filtered (note should not appear)
  - Feature flag is set to `false` (filtering should be disabled)
- [ ] **Check downstream telemetry consumers** - New keys `non_diff_filtered_count` and `non_diff_filtered_files` are added to `publish_result` state

**Recommended test plan:**
1. Set `REVIEWER_FILTER_NON_DIFF_FILES=true` (or use default)
2. Trigger a code review on a PR
3. Check logs for `[Publisher] EPIC B Optimization: Filtered X comments`
4. Verify the posted review includes the transparency note if comments were filtered

### Notes
- Blueprint Alignment: Telemetry v2 (observability), Diff-Aware Review (focus on changes), Feature Flag (controlled rollout)
- No unit tests added for `filter_non_diff_file_comments()` - consider adding in follow-up

Link to Devin run: https://app.devin.ai/sessions/45c687b0030a46e78cb181beee726167
Requested by: Ryan Chen (@RC918)

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/review_comment_schema.py [hunk 1]
```diff
@@ -1020,3 +1020,87 @@ def get_diff_coverage_info(
...
+        allowed_lines_map: Result from parse_diff_allowed_lines (files in diff)
+
+    Returns:
+        Tuple of (kept_comments, filtered_comments, filter_stats)
+        - kept_comments: Comments for files in the diff (or file-level without path)
+        - filtered_comments: Comments for files NOT in the diff
+        - filter_stats: Dict with telemetry data:
+          - filtered_count: Number of comments filtered
+          - kept_count: Number of comments kept
+          - filtered_files: Set of file paths that had comments filtered
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/review_comment_schema.py]: IMPACT: The use of a Set for filtered_files may not be compatible with all downstream systems that expect a list. This could cause issues if the telemetry or logging expects a list format.
SUGGESTED FIX: Consider changing the type to List[str] for consistency with other telemetry data structures.

File: handoff/20250928/40_App/orchestrator/review_comment_schema.py [hunk 2]
```diff
@@ -1020,3 +1020,87 @@ def get_diff_coverage_info(
...
+            filtered.append(comment)
+            filtered_files.add(file_path)
+
+    filter_stats = {
+        "filtered_count": len(filtered),
+        "kept_count": len(kept),
+        "filtered_files": list(filtered_files),
+        "filtered_file_count": len(filtered_files),
+    }
+
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/review_comment_schema.py]: IMPACT: When logging filtered_files, the code limits the output to 10 files. This could hide important information in cases where many files are filtered, potentially making it harder to diagnose issues.
SUGGESTED FIX: Consider increasing the limit or making it configurable, especially for debugging purposes.
